pkg/cmd/pulumi-language-java/main.go

```diff
@@ -138,9 +142,9 @@ func newLanguageHost(execOptions executors.JavaExecutorOptions,
 	}
 }
 
-func (host *javaLanguageHost) Executor() (*executors.JavaExecutor, error) {
-	if host.currentExecutor == nil {
-		executor, err := executors.NewJavaExecutor(host.execOptions)
+func (host *javaLanguageHost) Executor(attachDebugger bool) (*executors.JavaExecutor, error) {
+	if host.currentExecutor == nil || attachDebugger {
+		executor, err := executors.NewJavaExecutor(host.execOptions, attachDebugger)
```

Comment:
Should this be part of `execOptions`, maybe?

Reply:
I had it there initially, but it felt better to do it this way.  `execOptions` is a global thing, while we only want to attach the debugger when running the program (and only have access to the information whether a debugger is going to be attached when running the program).  So adding it to `execOptions` gets a bit messy.